
syntax = "proto3";
package pool;

import "common.proto";


service StatePool {

    rpc push_state(ServerState) returns (common.Empty);
    rpc pull_all_states(common.Empty) returns (StateMap);
}

message ServerState {
    string server_id = 1;
    string state = 2;
}

message StateMap {
    map<string, string> state_map = 1;
}

